packages/vite/src/assets.ts

```diff
@@ -0,0 +1,76 @@
+import type { Resolver } from '@embroider/core';
+import { ResolverLoader } from '@embroider/core';
+import type { Plugin } from 'vite';
+import * as process from 'process';
+import { join } from 'path';
+import { existsSync, readFileSync } from 'fs-extra';
+import send from 'send';
+import type { Readable } from 'stream';
+
+function findPublicAsset(relativePath: string, resolver: Resolver) {
+  const packageCache = resolver.packageCache;
+  let pkg = packageCache.ownerOfFile(relativePath);
+
+  for (const engine of resolver.options.engines) {
+    for (const addon of engine.activeAddons) {
+      pkg = packageCache.ownerOfFile(addon.root);
+      if (pkg && pkg.meta && pkg.isV2Addon() && pkg.meta['public-assets']) {
+        const asset = Object.entries(pkg.meta['public-assets']).find(([_key, a]) => a === relativePath)?.[0];
```

Comment:
Sure, maybe you can just push it to this pr?

Reply:
Will do 👍

Reply:
@BlueCutOfficial in the demo, did you try to build and open dist/index.html eith e.g local-web-server to check if it loads the assets?

Reply:
Also, I wonder if embroider changs src URLs, is vitr still able to load from app/public?

Reply:
Interesting. The new commits fixed the case of my classic addons but broke other test cases. We should have a deeper look when we have time.

Reply:
Maybe this is necessary:
https://github.com/ember-cli/ember-welcome-page/blob/33b4adc7b66fb5c656de46580b7424ba0576e671/ember-welcome-page/src/components/welcome-page.hbs#L4

Reply:
The demo addon I used to test public images uses `src` with and without  `{{this.rootURL}}`. I wanted to see if it makes a difference and I didn't notice any. Embroider generates the same path patterns in both cases 🤷‍♀️

Reply:
Ok, so I dropped the clean-up commits that remove the code copying the assets in the rewritten app and all tests are green again ✅ 🎉 

To me, it means that all tests failing when we have these commits present can potentially be deleted, because they rely on assets no longer present in the rewritten app. But it would probably be better to have a careful look at each of them, so a separate PR for the clean-up could be worth it, no strong opinion. I still think the clean-up is part of the functional piece, but no need to make this too big.